mlir/lib/IR/AffineExpr.cpp

```diff
@@ -239,6 +239,39 @@ bool AffineExpr::isPureAffine() const {
   llvm_unreachable("Unknown AffineExpr");
 }
 
+static bool isNonNegativeConstant(AffineExpr expr) {
+  auto constant = dyn_cast<AffineConstantExpr>(expr);
+  return constant && constant.getValue() >= 0;
+}
+
+bool AffineExpr::isMonotonic() const {
```

Comment:
```suggestion
bool AffineExpr::isMonotonicallyIncreasing() const {
```

If it is monotonically decreasing (only negative coefficients), then I guess this returns false.